articles/quickstart/backend/ruby/01-authentication.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Authentication
+description: This tutorial demonstrates how to add authentication to Ruby API
+---
+
+<%= include('../../../_includes/_package', {
+  org: 'auth0-samples',
+  repo: 'auth0-ruby-api-samples',
+  path: '01-Authenticate-RS256',
+  requirements: [
+    'Ruby 2.3.0'
+  ]
+}) %>
+
+<%= include('../_includes/_api_auth_preamble') %>
+
+This sample demonstrates how to check for a JWT in the `Authorization` header of an incoming HTTP request and verify that it is valid. The validity check is done using the **jwt** Gem within a custom `JsonWebToken` class. An `authenticate!` method is used to mark endpoints which require authentication through an incoming `access_token`. If the token is valid, the resources which are served by the endpoint can be released, otherwise a `401 Authorization` error will be returned.
+
+## Install the Dependencies
+
+Install the **jwt** Gem.
+
+```bash
+gem `jwt`
+bundle install
+```
+
+## Create a JsonWebToken Class
+
+<%= include('../_includes/_api_jwks_description', { sampleLink: 'https://github.com/auth0-samples/auth0-rubyonrails-api-samples/tree/OIDC/02-Authentication-HS256' }) %>
+
+Create a class called `JsonWebToken` which decodes and verifies the incoming `access_token` taken from the `Authorization` header of the request. The public key for your Auth0 tenant can be fetched to verify the token.
+
+```rb
+# lib/jwt/json_web_token.rb
+
+# frozen_string_literal: true
+require 'net/http'
+require 'uri'
+
+class JsonWebToken
+  def self.verify(token)
+    JWT.decode(token, nil,
+               true, # Verify the signature of this token
+               algorithm: 'RS256',
+               iss: Sinatra::Application.settings.auth0_domain,
```

Comment:
let's populate values dynamically where we can here. so for `iss`, let's have this in the code snippet (only here in the quickstart, not in the sample). These values come from the reader's Auth0 account if they are currently logged into the site.

```
iss: 'https://${account.namespace}/'
```